src/component/parser/xml/BpmnXmlParser.ts

```diff
@@ -44,4 +44,12 @@ export default class BpmnXmlParser {
     }
     return model;
   }
+
+  private doParsing(xml: string): BpmnJsonModel {
```

Comment:
❓ Why do you create another function just for the catch of the exception and not do it in the `parse`function ?

Reply:
Otherwise I have to do something like this. If you prefer this form, I can change the code.

```typescript
  parse(xml: string): BpmnJsonModel {
    let model;
    try {
      model = parse(xml, this.options);
    } catch {
      throw new Error('XML parsing failed. Invalid BPMN source.');
    }

    if (!model.definitions) {
        throw new Error(`XML parsing failed. Unable to retrieve 'definitions' for the BPMN source.`);
    }
    return model;
  }
```

Reply:
❓ And why not something like this ?
```suggestion
  parse(xml: string): BpmnJsonModel {
    try {
      const model = parse(xml, this.options);
      
      if (!model.definitions) {
        throw new Error(`XML parsing failed. Unable to retrieve 'definitions' for the BPMN source.`);
      }
      return model;
    } catch {
      throw new Error('XML parsing failed. Invalid BPMN source.', e);
    }
  }
```

Reply:
In this case, the first error is caught by the global catch, so the message is lost.
TypeScript doesn't like the Error chaining, ts-ignore must be used.
The code is not clearer.

Let's talk about it directly.

Reply:
Decision: implement the `let model` way as proposed in https://github.com/process-analytics/bpmn-visualization-js/pull/1752#discussion_r783972693